android-lib/src/main/java/com/kinvey/android/async/CallableAsyncPushRequestHelper.java

```diff
@@ -18,21 +20,34 @@
     private final SyncManager manager;
     private final SyncRequest syncRequest;
     private final SyncItem syncItem;
+    private final Class<T> storeItemType;
 
     CallableAsyncPushRequestHelper(AbstractClient client,
                            SyncManager manager,
                            SyncRequest syncRequest,
-                           SyncItem syncItem) {
+                           SyncItem syncItem,
+                           Class<T> storeItemType) {
         this.client = client;
         this.manager = manager;
         this.syncRequest = syncRequest;
         this.syncItem = syncItem;
+        this.storeItemType = storeItemType;
     }
 
     @Override
     public Long call() throws Exception {
         try {
-            manager.executeRequest(client, syncRequest);
+            if (syncRequest.getHttpVerb() == SyncRequest.HttpVerb.POST) {
+                String tempID = syncRequest.getEntityID().id;
+                GenericJson result = manager.executeRequest(client, syncRequest);
+                ICache<T> cache = client.getCacheManager().getCache(syncRequest.getCollectionName(), this.storeItemType, Long.MAX_VALUE);
```

Comment:
When we call `client.getCacheManager().getCache`, `cache.get`, `cache.delete`, `cache.save` the SDK opens and closes realm instance, so for creating 1 item we will open and close realm instance additionally 4 times. Opening/closing realm instance aren't quick operations. How will it affect performance?

Reply:
This will most likely affect performance, but this should be limited to the cases where we are creating entities locally, and not in general. Based on current design, this is necessary for correctness, but we can revisit the performance implications once we take a deeper look at the overall design.